src/test/test-varlink.c

```diff
@@ -205,6 +205,9 @@ static int overload_reply(sd_varlink *link, sd_json_variant *parameters, const c
 
         log_debug("Over reply triggered with error: %s", strna(error_id));
         ASSERT_STREQ(error_id, SD_VARLINK_ERROR_DISCONNECTED);
+        /* Local disconnect errors carry no parameters. Ensure we propagate
+         * absence as NULL rather than an empty object. */
+        ASSERT_TRUE(parameters == NULL);
```

Comment:
we'd usually do `ASSERT_TRUE(!parameters)` here, i.e. for pointers we use C's "everything downgrades to bool" feature.